ledger/block/src/transaction/deployment/mod.rs

```diff
@@ -57,12 +77,28 @@ impl<N: Network> Deployment<N> {
     pub fn check_is_ordered(&self) -> Result<()> {
         let program_id = self.program.id();
 
-        // Ensure the edition is either zero or one.
-        ensure!(
-            self.edition == 0 || self.edition == 1,
-            "Deployed the wrong edition (expected 0 or 1, found '{}').",
-            self.edition
-        );
+        // Ensure that either the both the program checksum and owner are present, or both are absent.
+        // The call to `Deployment::version` implicitly performs this check.
+        self.version()?;
+        // Validate the deployment based on the program checksum.
+        match self.program_checksum {
```

Comment:
Do we need to check that if the `checksum` exists, that the `constructor` must also exist?

Reply:
We don't enforce that since after V9, an upgrade for a program deployed during V8 may need to be made.
This also leaves room for additional migrations for non-V9 programs in the future.